modules/plugins/jboss-as-7/src/main/resources/META-INF/rhq-plugin.xml

```diff
@@ -5922,6 +5929,13 @@
           <c:simple-property name="module" required="false" type="string" readOnly="false" description="The module to load the object factory from for object factory bindings"/>
           <c:simple-property name="type" required="false" type="string" readOnly="false" description="The type of the value to bind for simple bindings, this must be a primitive type"/>
           <c:simple-property name="value" required="false" type="string" readOnly="false" description="The value to bind for simple bindings"/>
+          <c:simple-property name="cache" required="false" type="boolean" readOnly="false" defaultValue="false" description="If the external context should be cached"/>
+          <c:list-property name="environment" readOnly="true" description="A list of The environment to use on object factory instance retrieval." required="false">
+            <c:map-property name="environment" readOnly="true" displayName="Environment">
```

Comment:
why `readOnly=true` on the list and the map?

Reply:
My mistake, should be readonly= false,